pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -163,7 +156,15 @@ func (p *Predictor) Reconcile(isvc *v1beta1.InferenceService) error {
 		} else {
 			podSpec.Containers[0] = *container
 		}
+	}
 
+	// Knative does not support INIT containers or mounting, so we add annotations that trigger the
+	// StorageInitializer injector to mutate the underlying deployment to provision model data
+	if sourceURI := predictor.GetStorageUri(); sourceURI != nil {
+		if _, ok := annotations[constants.StorageInitializerSourceUriInternalAnnotationKey]; ok {
+			return errors.New("must provide only one of storageUri and storage.path")
```

Comment:
Not sure if I understand this check here.

Reply:
If storageSpec with path exist, the `addStorageSpecAnnotations` function at line 74 will set the uri annotation. Therefore if the storageURI is defined again, we will throw an error because there's conflict when users define both storageSpec.path and storageURI.